MovieDB/DI/Composers/MainComposer.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MainComposer.swift
+//  MovieDB
+//
+//  Created by Mouad Bj on 4/7/2023.
+//
+
+import UIKit
+import MovieDBCore
+import MovieDBUI
+
+// MARK: - Main Dependency Composer
+class MainComposer {
+  
+  let container: DIContainer
+  var coordinator: AppCoordinator?
+  
+  init(container: DIContainer) {
+    self.container = container
+  }
+  
+  func startApp(with navigationController: UINavigationController) {
+    coordinator = makeAppCoordinator(navigationController: navigationController)
+    coordinator?.start()
+  }
+  
+  func makeAppCoordinator(navigationController: UINavigationController) -> AppCoordinator {
+    return AppCoordinatorMain(navigationController: navigationController, dependencyFactory: self)
+  }
+  
+}
+
+// MARK: - AppCoordinator Dependency Factory
+extension MainComposer: AppCoordinatorDependencyFactory {
+
+  func makeMoviesListScene() -> UIViewController? {
+    guard let moviesInteractor = try? container.resolve(.Domain.Interactor.movies) ,
+          let analyticsManager = try? container.resolve(.Domain.Interactor.analytics) else { return nil }
```

Comment:
- Totally agree 👍🏽, the DI Container I wrote is just a simple implementation for showcase, not a robust production one, (As my purpose with this code is to showcase the high-level separation of modules and not low-level implementation details [as important as they are]), I also wanted to leave the sample code -at this stage- free of third-party libraries as I think choosing the appropriate DI manager is a team discussion.
- I will check the two options u proposed, I am sure they will have better management for these cases 👍🏽. 
- if we want to fix the mentioned issues, we can do force unwrapping within the DI container so it will cause a crash if a certain dependency is not registered. ideally throw an error with an appropriate error message for debugging (witch, I am sure, will be well managed in a more robust production DI solution)